dlib/media/ffmpeg_utils.h

```diff
@@ -394,33 +439,206 @@ namespace dlib
             }
         }
 
-        inline std::string get_pixel_fmt_str(AVPixelFormat fmt)
+// ---------------------------------------------------------------------------------------------------
+
+        ///////////////////////////
+        // Channel layout stuff 
+        ///////////////////////////
+
+        inline uint64_t get_layout_from_channels(const std::size_t nchannels)
```

Comment:
`get_layout_from_channels`, `get_channels_from_layout`, and some of the other stuff below isn't mentioned in the abstract.h docs file.  Should they be?  Seems like these are meant to be hidden implementation details?  Can they be in `details`?

Reply:
Will check